tools/update-npm.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+set -e
+# Shell script to update npm in the source tree to a specific version
+
+BASE_DIR="$( pwd )"/
+DEPS_DIR="$BASE_DIR"deps/
+NPM_VERSION=$1
+
+if [ "$#" -le 0 ]; then
+  echo "Error: please provide an npm version to update to"
+  exit 1
+fi
+
+WORKSPACE="$TMPDIR"update-npm-$NPM_VERSION/
+
+echo "Making temporary workspace"
+
+mkdir -p $WORKSPACE
```

Comment:
On a sode note, there are tools that do static analysis of shell scripts that we might want to use to "lint" them in core (E.G.: https://github.com/koalaman/shellcheck).

Reply:
Yeah, I do `npx shellcheck <file>` a lot and have been wanting to add it to Node.js linting, but it would be a *lot* of churn. If someone else wants to make it happen, I will happily 👍 the work, though.

Reply:
fixed